### ISSUE TITLE
[Bug]: Keyboard shortcuts cannot be changed by keyboard users

### ISSUE THREAD

**Issue by Sidnioulz** [storybookjs/storybook#32311]
### Describe the bug

When using the keyboard shortcuts page (http://localhost:6006/?path=/settings/shortcuts), it's impossible to navigate through the list without overwriting shortcuts.

Specifically, pressing Shift+Tab to navigate to the previous shortcut will cause the currently focused shortcut to be overwritten by Shift.

### Reproduction link

yarn storybook:ui

### Reproduction steps

1. Go to http://localhost:6006/?path=/settings/shortcuts
2. Tab into a shortcut
3. Press Shift+Tab to tab backwards
4. Notice the shortcut is overwritten

### System

```bash
10.0 beta 0
```
